homeassistant/components/plugwise/sensor.py

```diff
@@ -0,0 +1,429 @@
+        ENERGY_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_consumed_off_peak_point": [
+        "Current Consumed Power (off peak)",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_consumed_peak_point": [
+        "Current Consumed Power",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:flash",
+    ],
+    "electricity_consumed_off_peak_cumulative": [
+        "Cumulative Consumed Power (off peak)",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+    "electricity_consumed_peak_cumulative": [
+        "Cumulative Consumed Power",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+    "electricity_produced_off_peak_point": [
+        "Current Consumed Power (off peak)",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:white-balance-sunny",
+    ],
+    "electricity_produced_peak_point": [
+        "Current Consumed Power",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:white-balance-sunny",
+    ],
+    "electricity_produced_off_peak_cumulative": [
+        "Cumulative Consumed Power (off peak)",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+    "electricity_produced_peak_cumulative": [
+        "Cumulative Consumed Power",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+    "gas_consumed_interval": [
+        "Current Consumed Gas",
+        VOLUME_CUBIC_METERS,
+        None,
+        "mdi:gas-cylinder",
+    ],
+    "gas_consumed_cumulative": [
+        "Cumulative Consumed Gas",
+        VOLUME_CUBIC_METERS,
+        None,
+        "mdi:gauge",
+    ],
+    "net_electricity_point": [
+        "Current net Power",
+        POWER_WATT,
+        DEVICE_CLASS_POWER,
+        "mdi:solar-power",
+    ],
+    "net_electricity_cumulative": [
+        "Cumulative net Power",
+        ENERGY_KILO_WATT_HOUR,
+        DEVICE_CLASS_POWER,
+        "mdi:gauge",
+    ],
+}
+
+MISC_SENSOR_MAP = {
+    "battery": ATTR_BATTERY_LEVEL,
+    "illuminance": ATTR_ILLUMINANCE,
+    "modulation_level": [
+        "Heater Modulation Level",
+        UNIT_PERCENTAGE,
+        "modulation",
+        "mdi:percent",
+    ],
+    "valve_position": ["Valve Position", UNIT_PERCENTAGE, None, "mdi:valve"],
+    "water_pressure": ATTR_PRESSURE,
+}
+
+INDICATE_ACTIVE_LOCAL_DEVICE = [
+    "cooling_state",
+    "flame_state",
+]
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Smile sensors from a config entry."""
+    api = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
+
+    entities = []
+    all_devices = api.get_all_devices()
+    single_thermostat = api.single_master_thermostat()
+    for dev_id, entity in all_devices.items():
```

Comment:
```suggestion
    for dev_id, device_properties in all_devices.items():
```

These aren't entities right?  These are the properties (naming?) that are going to get passed to `PwPowerSensor` or similar to create the entities.

Reply:
Changed accordingly but cleaned up in climate as well for consistency